src/Nri/Ui/FocusLoop/V1.elm

```diff
@@ -1,80 +1,76 @@
-module Nri.Ui.FocusLoop.V1 exposing (addEvents)
+module Nri.Ui.FocusLoop.V1 exposing (Config, view, addEvents)
 
-{-| Sometimes, there are sets of interactive elements that we want users to be able to navigate through with arrow keys rather than with tabs, and we want the final focus change to wrap. This module makes it easier to set up this focus and wrapping behavior.
+{-| Sometimes, there are sets of interactive elements that we want users to be able to navigate
+through with arrow keys rather than with tabs, and we want the final focus change to wrap.
+This module makes it easier to set up this focus and wrapping behavior.
 
-@docs addEvents
+@docs Config, view, addEvents
 
 -}
 
-import Accessibility.Styled.Key as Key exposing (Event)
+import Accessibility.Styled exposing (Html)
+import Accessibility.Styled.Key as Key
+import Nri.Ui.FocusLoop.Internal exposing (keyEvents, siblings)
 
 
-{-| -}
-addEvents :
-    { focus : a -> msg
+{-| FocusLoop.view Configuration
+-}
+type alias Config id msg args =
+    { toId : args -> id
+    , focus : id -> msg
+    , view : args -> List (Key.Event msg) -> Html msg
     , leftRight : Bool
     , upDown : Bool
     }
-    -> List a
-    -> List ( a, List (Event msg) )
-addEvents config items =
-    case items of
-        [] ->
-            []
 
-        item :: [] ->
-            [ ( item, [] ) ]
 
-        _ ->
-            addEvents_ config items
+{-| Helper for creating a list of elements with looping arrow-key navigation.
+
+Your `view` function will be called for each item with the corresponding keyboard
+event handlers, as well as the item itself.
+
+e.g.
+
+    FocusLoop.view
+        { id = .id
+        , focus = Focus
+        , leftRight = True
+        , upDown = True
+        , view = viewFocusableItem
+        }
+        [ { id = 1, name = "Foo" } ]
+
+    viewFocusableItem item arrowKeyHandlers =
+        div
+            [ Key.onKeyDownPreventDefault arrowKeyHandlers ]
+            [ text item.name ]
 
+Does your list support adding and removing items? If so, check out `FocusLoop.Lazy` which
+will prevent recalculation of event handlers for every item when the list changes.
 
-addEvents_ :
+-}
+view : Config id msg item -> List item -> List (Html msg)
```

Comment:
We should avoid type alias of records in exported functions to have better error reporting https://github.com/NoRedInk/NoRedInk/blob/master/docs/sdgg/elm-style-guide.md#prefer-explicit-types-in-type-signatures-to-type-aliases

Reply:
hmm, if we're going to do that we can drop the uncurry from the view and id too and pass those args in separately. the only reason I didn't do that was trying to avoid having a bunch of different config types

Reply:
Sounds good to drop the uncurry and have specific record types for view/view2/viewN. If we need to keep the Args type maybe that can be an internal not exported type.

I would not implement all N, just up to the one we need to student interest.

Reply:
Playing with this now, I'm not sure if we can drop the uncurry _and_ keep everything contained to a helper fn without higher-kinded types. I'd prefer to avoid having the business logic in the fn for each arity.

> I would not implement all N, just up to the one we need to student interest.

Any particular reason? I've found this can change/fluctuate pretty rapidly as functionality is added, and since we can only have up to 5 anyway idk why we wouldn't have all of them from the get-go.